lib/core/src/lib/services/discovery-api.service.ts

```diff
@@ -42,7 +42,10 @@ export class DiscoveryApiService {
     ) {
         this.authenticationService.onLogin
             .pipe(
-                filter(() => this.apiService.getInstance()?.isEcmLoggedIn()),
+                filter(() => this.apiService.getInstance()?.isEcmLoggedIn() ||
+                ((this.authenticationService.isECMProvider() ||
```

Comment:
This condition seems to be redundant

Reply:
Why do you think so? In case of `isEcmLoggedIn` returns `false` we check for ecm or all provider but in this case Kerberos must be enabled as well.

Reply:
have you check isEcmLoggedIn inside? does some of the thing you are adding here.

probably will make more sense move this methods in the authentication 

https://github.com/Alfresco/alfresco-ng2-components/blob/147fb5672448dbdb51296c005fcde8b292749411/lib/core/src/lib/userinfo/components/user-info.component.ts#L139 

and use it?

or would be better implement this logic before to do the next in the onLogin ?

https://github.com/Alfresco/alfresco-ng2-components/blob/4410557d8dbf6573d430c3cda7d35330d5975c00/lib/core/src/lib/services/authentication.service.ts#L73

Otherwise all the call that use  needs onLogin will need to add those checks

Reply:
@eromano I did, but this is different case I think, when using Kerberos, `isEcmLoggedIn()` always returns false and those checks are added for discoveryService to be able to retrieve the ECM product info with Kerberos enabled. Also I don't think that we need to modify each subscription to `onLogin` but I don't know this area that well. If you're unsure about this solution or if you think that it should be added in some other areas please talk to Nicolas Barithel. He is the author of the change but I recreated his PR so we can unit test the changes, the original PR -> https://github.com/Alfresco/alfresco-ng2-components/pull/7851. In case of any changes that has to be made please let me know.

Reply:
@eromano could you please check out my comment?